File: runround/runround.py

"""
ID: heng_li1
LANG: PYTHON2
TASK: runround
"""
fin = open ('runround.in', 'r')
fout = open ('runround.out', 'w')

n = int(fin.readline())

def digits(n):
	return map(int, str(n))

dn = digits(n)
if 0 in dn:
	ind = dn.index(0)
	while ind < len(dn):
		dn[ind] = 1
		ind +=1

n = int(''.join(map(str, dn)))

def incrment(n):
	q,r = divmod(n,10)
	if r<9:
		return n+1
	return incrment(q)*10+1



def runround(n):
	dn = digits(n)
	curr = 0
	seq = []
	seen = [False]*10
	while len(seq)<len(dn):
		seq.append(dn[curr])
		seen[dn[curr]] = True
		curr += dn[curr]
		curr %= len(dn)
		check = dn[curr]!=seq[0] if len(seq)==len(dn) else seen[dn[curr]]
		if check:
			return False
	return True



n=incrment(n)
while not runround(n):
	n=incrment(n)

print >>fout, n

fout.close()

File: runround/runround.java

/*
ID: heng_li1
LANG: JAVA
TASK: runround
*/
import java.io.*;
import java.util.*;

public class runround {
  public static void main (String [] args) throws IOException {
    // Use BufferedReader rather than RandomAccessFile; it's much faster
    BufferedReader f = new BufferedReader(new FileReader("runround.in"));
                                                  // input file name goes above
    PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("runround.out")));
    // Use StringTokenizer vs. readLine/split -- lots faster
    StringTokenizer st = new StringTokenizer(f.readLine());
						  // Get line, break into tokens
    
    out.close();                                  // close the output file
  }
}
